dashboard/legacy/middleware/helpers/projects.rb

```diff
@@ -488,6 +488,7 @@ def validate_thumbnail_url(channel_id, thumbnail_url)
   # I observed thumbnail URLs of remixed projects having having the channel ID of the parent project,
   # so we assert on the start/end of the URL
   def valid_thumbnail_url?(thumbnail_url)
-    thumbnail_url.start_with?('/v3/files/') && thumbnail_url.end_with?('.metadata/thumbnail.png')
+    (thumbnail_url.start_with?('/v3/files/') && thumbnail_url.end_with?('.metadata/thumbnail.png')) ||
```

Comment:
This is the moment when I feel proud of my comment here :D Keeping it clean <3 

https://github.com/code-dot-org/code-dot-org/pull/51981#discussion_r1200904550


if the list keeps growing, we might want to move this to a regular expression list.
We can have a global variable with all those regular expressions and validate based on that.

Overkill for only 2 rules for now. Just adding the comment since you know how the list will grow and if to see if it is a good idea for the future.

Reply:
haha it was a good call! Yeah I am hopeful these are the only two cases for now. LMK if this change looks ok to you.

Reply:
Can we get away with simpler rules? (Is the `end_with` necessary?)
Are we requiring that these be urls within our domain?

While I'm here, the regex is pretty gnarly, so I personally think the readability of `start_with?` outweighs the possible performance gain of a regex.

```ruby
pattern = /^\/v3\/files\/\w+\/\.metadata\/thumbnail\.png$|^\/blockly\/media/
thumbnail_url.match?(pattern)
```

Reply:
> Are we requiring that these be urls within our domain?

I think so? Are you suggesting that is too restrictive, or that having a leading slash doesn't actually guarantee that they're within our domain? Feel free to follow up in Slack if that's a more appropriate forum for this discussion.

FWIW, the example case in the Jira ticket uses a relative path in a way that would produce negative behavior, so I don't think I can just say "any relative path is valid".

Reply:
as we discussed, i'm on the same page now! The prefix check will ensure that these are always relative urls, and therefore preventing other domains from being saved :+1: